Alignment/TrackerAlignment/plugins/CosmicRateAnalyzer.cc

```diff
@@ -0,0 +1,490 @@
+   treeEvent	->Branch("ntrk",&ntrk);
+   treeEvent	->Branch("DTtime",&DTtime);
+   treeRun	->Branch("run_time",&run_time);   
+   treeRun	->Branch("runnum",&runnum);   
+   treeRun	->Branch("number_of_events",&number_of_events);
+   treeRun	->Branch("number_of_tracks",&number_of_tracks);
+   treeRun	->Branch("number_of_tracks_PIX",&number_of_tracks_PIX);
+   treeRun	->Branch("number_of_tracks_FPIX",&number_of_tracks_FPIX);
+   treeRun	->Branch("number_of_tracks_BPIX",&number_of_tracks_BPIX);
+   treeRun	->Branch("number_of_tracks_TID",&number_of_tracks_TID);
+   treeRun	->Branch("number_of_tracks_TIDM",&number_of_tracks_TIDM);
+   treeRun	->Branch("number_of_tracks_TIDP",&number_of_tracks_TIDP);
+   treeRun	->Branch("number_of_tracks_TIB",&number_of_tracks_TIB);
+   treeRun	->Branch("number_of_tracks_TEC",&number_of_tracks_TEC);
+   treeRun	->Branch("number_of_tracks_TECP",&number_of_tracks_TECP);
+   treeRun	->Branch("number_of_tracks_TECM",&number_of_tracks_TECM);
+   treeRun	->Branch("number_of_tracks_TOB",&number_of_tracks_TOB);
+   treeCluster	->Branch("DetID",&DetectorID);
+
+}
+
+// ------------ method called once each job just after ending the event loop  ------------
+void 
+CosmicRateAnalyzer::endJob() 
+{
+}
+
+// ------------ method called when starting to processes a run  ------------
+void 
+CosmicRateAnalyzer::beginRun(edm::Run const&, edm::EventSetup const&)
+{
+   lastruntime		= 0.0;
+   lastrunnum		= 0.0;
+   ntrk_runnum 		= 0.0;
+   events		= 0.0;
+   track_BPIX           = 0.0;  
+   track_FPIX           = 0.0;
+   track_PIXEL          = 0.0;
+   track_TEC            = 0.0;
+   track_TECM           = 0.0;
+   track_TECP           = 0.0;
+   track_TOB            = 0.0;
+   track_TIB            = 0.0;
+   track_TID            = 0.0;
+   track_TIDM           = 0.0;
+   track_TIDP           = 0.0;
+}
+
+
+// ------------ method called when ending the processing of a run  ------------
+
+void 
+CosmicRateAnalyzer::endRun(edm::Run const&, edm::EventSetup const&)
+{
+   number_of_tracks	=ntrk_runnum;
+   run_time		= lastruntime;
+   runnum		= lastrunnum;
+   number_of_tracks_PIX	= track_PIXEL ; 
+   number_of_tracks_FPIX= track_FPIX ; 
+   number_of_tracks_BPIX= track_BPIX; 
+   number_of_tracks_TEC	= track_TEC  ; 
+   number_of_tracks_TECM= track_TECM ; 
+   number_of_tracks_TECP= track_TECP ; 
+   number_of_tracks_TOB	= track_TOB  ; 
+   number_of_tracks_TIB	= track_TIB  ; 
+   number_of_tracks_TID	= track_TID  ; 
+   number_of_tracks_TIDM= track_TIDM ; 
+   number_of_tracks_TIDP= track_TIDP ; 
+   number_of_events	= events;
+   treeRun  ->Fill();
+}
+
+// ------------ method fills 'descriptions' with the allowed parameters for the module  ------------
+void
+CosmicRateAnalyzer::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  //The following says we do not know what parameters are allowed so do no validation
+  // Please change this to state exactly what you do use, even if it is no parameters
```

Comment:
@skeshri I just realized, that you should remove this comment, now that you have a proper description in place